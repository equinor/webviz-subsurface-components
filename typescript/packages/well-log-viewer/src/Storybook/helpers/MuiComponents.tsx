import React, { useEffect } from "react";

import type { TooltipProps as MuiTooltipProps } from "@mui/material";
import {
    Box,
    // eslint-disable-next-line no-restricted-imports
    Tab,
    // eslint-disable-next-line no-restricted-imports
    Tabs as MUITabs,
    Tooltip as MuiToolTip,
    Typography,
    tabsClasses,
} from "@mui/material";

/**
 * Remove spaces and set to lower case.
 * Note that we cannot use replaceAll function
 * below until we move to ES2021.
 */
const buildId = (id: string): string => {
    return id.replace(/ /g, "").toLowerCase();
};

type TooltipProps = Pick<
    MuiTooltipProps,
    "children" | "title" | "placement" | "id" | "followCursor"
> & { disable?: boolean; fullWidth?: boolean };

const Tooltip: React.FC<TooltipProps> = (props) => {
    const { children, disable = false, ...rest } = props;
    const childrenWrapper = disable ? <span>{children}</span> : children;
    const ref = React.useRef<HTMLDivElement>();
    const childrenElement = React.cloneElement(childrenWrapper, { ref: ref });
    const [skipTooltip, setSkipTooltip] = React.useState(false);
    const [open, setOpen] = React.useState(false);

    React.useEffect(() => {
        // If a tooltip is placed on an element that could potentially overflow and is truncated with ellipsis,
        // we only show the tooltip in the case where the element is actually overflowing
        const textNode = ref.current as HTMLElement;
        if (
            textNode &&
            textNode.clientWidth >= textNode.scrollWidth &&
            textNode.clientWidth > 0 &&
            textNode.innerText === props.title
        ) {
            setSkipTooltip(true);
        }
    }, [props.title, ref, childrenElement]);

    const handleClose = React.useCallback(() => {
        setOpen(false);
    }, []);

    const handleOpen = React.useCallback(() => {
        setOpen(true);
    }, []);

    if (skipTooltip) {
        return childrenElement;
    }

    return (
        <MuiToolTip
            placement={props.placement ?? "top-start"}
            enterDelay={500}
            open={open && !disable}
            onClose={handleClose}
            onOpen={handleOpen}
            disableFocusListener={disable}
            disableHoverListener={disable}
            disableTouchListener={disable}
            arrow
            {...rest}
        >
            {childrenElement}
        </MuiToolTip>
    );
};

/**
 * Defines the content of a Tab title
 * @param label: the tab title
 * @param data-tut: optional flag for guided-tour usage (see UserAssistance component)
 * @param attention: optional flag to highlight this tab
 * @param decoration: optional element to add on the right of the tab panel
 * @param tooltip: optional tooltip
 */
type TabTitle = {
    label: string;
    "data-tut"?: string;
    attention?: boolean;
    decoration?: JSX.Element;
    tooltip?: string;
};

/**
 * Defines the properties of the Tabs Bar component
 * @param currentTab: defines the selected Tab
 * @param name: defines the component name
 * @param onTabChange: function called when the value of the selected tab has changed
 * @param titles: describes the Tabs of the component
 */
type TabsBarProps = {
    currentTab: number;
    name: string;
    // eslint-disable-next-line @typescript-eslint/ban-types
    onTabChange: (e: React.ChangeEvent<{}>, newValue: number) => void;
    titles: TabTitle[];
};

/**
 * Convenient component to gather and control a list of Tabs
 * @param props the component properties
 * @returns the created TabsBar component
 */
const TabsBar: React.FC<TabsBarProps> = (props) => {
    if (props.titles.length === 0) {
        return null;
    }

    const tabs = props.titles.map((title, index) => (
        <Tooltip key={title.label} title={title.tooltip ?? ""}>
            <Tab
                label={
                    <Box
                        data-tut={title["data-tut"]}
                        id={`${props.name}-tablabel-${index}`}
                    >
                        <Typography
                            variant="body2" // TODO: should be changed to h5 when removing the (v1.5) theme
                            sx={{
                                display: "flex",
                                alignItems: "center",
                                gap: (theme) => theme.spacing(0.5),
                            }}
                        >
                            {title.label}
                        </Typography>
                        {title.decoration}
                    </Box>
                }
                {...accessibilityProps(index, props.name)}
                wrapped
            />
        </Tooltip>
    ));

    return (
        <MUITabs
            value={props.currentTab}
            onChange={props.onTabChange}
            indicatorColor="primary"
            textColor="primary"
            variant="scrollable"
            scrollButtons="auto"
            aria-label={`${props.name}-tabs`}
            sx={{
                [`& .${tabsClasses.scrollButtons}`]: {
                    "&.Mui-disabled": { opacity: 0.3 },
                },
            }}
        >
            {tabs}
        </MUITabs>
    );
};

/**
 * Convenient function to accessibility properties of a controlled Tab
 * by a TabsBar component.
 * @param index the Tab index in the TabsBar component
 * @param name the Tab name
 * @returns the Tab accessibility properties
 */
const accessibilityProps = (
    index: number,
    name: string
): { id: string; "aria-controls": string } => {
    return {
        id: `${name}-tab-${index}`,
        "aria-controls": `${name}-tabpanel-${index}`,
    };
};

/**
 * Defines the properties of the TabPanel component
 * @param name: the name of the TabPanel.
 * @param display: if true, the panel will be visible. Hidden otherwise.
 * @param customClassName: optional class name to control the component style.
 * @param children: optional children.
 */
type TabPanelProps = {
    name: string;
    display: boolean;
    customClassName?: string;
    children?: React.ReactNode;
};

/**
 * Convenient component to define a Tab panel which visibility is controlled by
 * a prop. Mainly designed for an association with the TabsBar component that will
 * control the visibility of a list of Tab panels in function of the selected tab.
 * @param props the component properties
 * @returns the created TabPanel component
 */
const TabPanel: React.FC<TabPanelProps> = (props) => {
    return (
        <div
            role="tabpanel"
            hidden={!props.display}
            id={`${buildId(props.name)}-tabpanel`}
            data-testid={`${buildId(props.name)}-tabpanel`}
            aria-labelledby={`${buildId(props.name)}-tab`}
        >
            {props.children}
        </div>
    );
};

/**
 * Defines a tab and its associated panel
 * @param title the tab title
 * @param panelContent defines the content of the tab panel
 * @param panelCustomClassName optional class name to control the tab panel style
 */
export type TabItem = {
    title: TabTitle;
    panelContent: JSX.Element;
    panelCustomClassName?: string;
};

/**
 * Defines the properties of the Tabs component
 * @param name the collection name. It will be used to tag the tab and panel
 * components contained in the collection
 * @param tabs defines the list of tabs and their panel contained in this collection
 * @param width defines the width of a collection inside a group.
 * @param selectedTab option number stating the tab that will be selected on build,
 * default is 0.
 * @param handleSelectedTab function that is called when new tab is selected.
 * @param groupIndex is the index of the current collection inside a group.
 */
type TabsProps = {
    name: string;
    tabs: TabItem[];
    width?: string | number;
    selectedTab?: number;
    handleSelectedTab?: (newSelectedTab: number) => void;
};

/**
 * Builds a tab panel
 * @param props: props of the panel to create.
 * @returns the created panel.
 */
const BuildTabPanel: React.FC<TabPanelProps> = React.memo((props) => {
    return (
        <TabPanel
            key={`${buildId(props.name)}-panel`}
            name={props.name}
            display={props.display}
            customClassName={props.customClassName}
        >
            {props.children}
        </TabPanel>
    );
});

BuildTabPanel.displayName = "BuildTabPanel";

/**
 * A convenient component to build a collection of tabs and their associated panels
 * @param props the component properties
 * @returns the created component or null if the props tabs array is empty
 */
export const Tabs: React.FC<TabsProps> = (props) => {
    const [currentTab, setCurrentTab] = React.useState(
        props.selectedTab ? props.selectedTab : 0
    );

    useEffect(() => {
        setCurrentTab(props.selectedTab ? props.selectedTab : 0);
    }, [props]);

    // eslint-disable-next-line @typescript-eslint/ban-types
    const handleChange = (_e: React.ChangeEvent<{}>, newValue: number) => {
        setCurrentTab(newValue);
        if (props.handleSelectedTab) {
            props.handleSelectedTab(newValue);
        }
    };

    if (props.tabs.length === 0) {
        return null;
    }

    return (
        <Box style={props.width ? { width: props.width } : undefined}>
            <TabsBar
                name={props.name}
                currentTab={currentTab}
                onTabChange={handleChange}
                titles={props.tabs.map((tab) => tab.title)}
            />
            {props.tabs.map((tab, index) => (
                <BuildTabPanel
                    key={`${buildId(tab.title.label)}-panel`}
                    display={index === currentTab}
                    name={tab.title.label}
                    customClassName={tab.panelCustomClassName}
                >
                    {tab.panelContent}
                </BuildTabPanel>
            ))}
        </Box>
    );
};

export default Tabs;
