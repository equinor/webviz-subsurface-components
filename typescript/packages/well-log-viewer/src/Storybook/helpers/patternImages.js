import Anhydrite from "../../../../../../example-data/patterns/Anhydrite.gif";
import Bitumenious from "../../../../../../example-data/patterns/Bitumenious.gif";
import Browncoal from "../../../../../../example-data/patterns/Browncoal.gif";
import CalcareousDolostone from "../../../../../../example-data/patterns/Calcareous_dolostone.gif";
import Chalk from "../../../../../../example-data/patterns/Chalk.gif";
import Clay from "../../../../../../example-data/patterns/Clay.gif";
import Coal from "../../../../../../example-data/patterns/Coal.gif";
import Conglomerate from "../../../../../../example-data/patterns/Conglomerate.gif";
import DiamondLines from "../../../../../../example-data/patterns/Diamond_lines.gif";
import DolomiticLimestone from "../../../../../../example-data/patterns/Dolomitic_limestone.gif";
import Dolostone from "../../../../../../example-data/patterns/Dolostone.gif";
import DownwardLines from "../../../../../../example-data/patterns/Downward_lines.gif";
import DykesAndSills from "../../../../../../example-data/patterns/Dykes_and_sills.gif";
import EmptyFile from "../../../../../../example-data/patterns/EmptyFile.gif";
import FissileMud from "../../../../../../example-data/patterns/Fissile_mud.gif";
import FissileSilt from "../../../../../../example-data/patterns/Fissile_silt.gif";
import GridLines from "../../../../../../example-data/patterns/Grid_lines.gif";
import Gypsum from "../../../../../../example-data/patterns/Gypsum.gif";
import GypsumAnhydriteUnspecified from "../../../../../../example-data/patterns/Gypsum_anhydrite_unspecified.gif";
import Halite from "../../../../../../example-data/patterns/Halite.gif";
import HorizontalDashed from "../../../../../../example-data/patterns/Horizontal_dashed.gif";
import HorizontalLines from "../../../../../../example-data/patterns/Horizontal_lines.gif";
import Intrusive from "../../../../../../example-data/patterns/Intrusive.gif";
import Limestone from "../../../../../../example-data/patterns/Limestone.gif";
import MaficPlutonic from "../../../../../../example-data/patterns/Mafic_plutonic.gif";
import Marl from "../../../../../../example-data/patterns/Marl.gif";
import Metamorphic from "../../../../../../example-data/patterns/Metamorphic.gif";
import Mud from "../../../../../../example-data/patterns/Mud.gif";
import Raster from "../../../../../../example-data/patterns/Raster.gif";
import SaltGeneral from "../../../../../../example-data/patterns/Salt_general.gif";
import Sand from "../../../../../../example-data/patterns/Sand.gif";
import SedimentBreccia from "../../../../../../example-data/patterns/Sediment_breccia.gif";
import Shale from "../../../../../../example-data/patterns/Shale.gif";
import SilicicPlutonic from "../../../../../../example-data/patterns/Silicic_plutonic.gif";
import Silt from "../../../../../../example-data/patterns/Silt.gif";
import Tuffitt from "../../../../../../example-data/patterns/Tuffitt.gif";
import UpwardLines from "../../../../../../example-data/patterns/Upward_lines.gif";
import VerticalBitumenious from "../../../../../../example-data/patterns/Vertical_bitumenious.gif";
import VerticalCalcareousDolostone from "../../../../../../example-data/patterns/Vertical_calcareous_dolostone.gif";
import VerticalChalk from "../../../../../../example-data/patterns/Vertical_chalk.gif";
import VerticalClaystone from "../../../../../../example-data/patterns/Vertical_claystone.gif";
import VerticalDashed from "../../../../../../example-data/patterns/Vertical_dashed.gif";
import VerticalDolomiticLimestone from "../../../../../../example-data/patterns/Vertical_dolomitic_limestone.gif";
import VerticalDolostone from "../../../../../../example-data/patterns/Vertical_dolostone.gif";
import VerticalFissileMudstone from "../../../../../../example-data/patterns/Vertical_fissile_mudstone.gif";
import VerticalFissileSiltstone from "../../../../../../example-data/patterns/Vertical_fissile_siltstone.gif";
import VerticalLimestone from "../../../../../../example-data/patterns/Vertical_limestone.gif";
import VerticalLines from "../../../../../../example-data/patterns/Vertical_lines.gif";
import VerticalMarl from "../../../../../../example-data/patterns/Vertical_marl.gif";
import VerticalShale from "../../../../../../example-data/patterns/Vertical_shale.gif";
import VerticalTuffitt from "../../../../../../example-data/patterns/Vertical_tuffitt.gif";
import Vulcanic from "../../../../../../example-data/patterns/Vulcanic.gif";

const imagesAndNames = Object.freeze([
    { img: Anhydrite, name: "Anhydrite" },
    { img: Bitumenious, name: "Bitumenious" },
    { img: Browncoal, name: "Browncoal" },
    { img: CalcareousDolostone, name: "Calcareous Dolostone" },
    { img: Chalk, name: "Chalk" },
    { img: Clay, name: "Clay" },
    { img: Coal, name: "Coal" },
    { img: Conglomerate, name: "Conglomerate" },
    { img: DiamondLines, name: "Diamond_lines" },
    { img: DolomiticLimestone, name: "Dolomitic_limestone" },
    { img: Dolostone, name: "Dolostone" },
    { img: DownwardLines, name: "Downward Lines" },
    { img: DykesAndSills, name: "Dykes and Sills" },
    { img: EmptyFile, name: "EmptyFile" },
    { img: FissileMud, name: "Fissile Mud" },
    { img: FissileSilt, name: "Fissile Silt" },
    { img: GridLines, name: "Grid Lines" },
    { img: Gypsum, name: "Gypsum" },
    { img: GypsumAnhydriteUnspecified, name: "Gypsum Anhydrite Unspecified" },
    { img: Halite, name: "Halite" },
    { img: HorizontalDashed, name: "Horizontal Dashed" },
    { img: HorizontalLines, name: "Horizontal Lines" },
    { img: Intrusive, name: "Intrusive" },
    { img: Limestone, name: "Limestone" },
    { img: MaficPlutonic, name: "Mafic Plutonic" },
    { img: Marl, name: "Marl" },
    { img: Metamorphic, name: "Metamorphic" },
    { img: Mud, name: "Mud" },
    { img: Raster, name: "Raster" },
    { img: SaltGeneral, name: "Salt General" },
    { img: Sand, name: "Sand" },
    { img: SedimentBreccia, name: "Sediment Breccia" },
    { img: Shale, name: "Shale" },
    { img: SilicicPlutonic, name: "Silicic Plutonic" },
    { img: Silt, name: "Silt" },
    { img: Tuffitt, name: "Tuffitt" },
    { img: UpwardLines, name: "Upward lines" },
    { img: VerticalBitumenious, name: "Vertical Bitumenious" },
    { img: VerticalCalcareousDolostone, name: "Vertical Calcareous Dolostone" },
    { img: VerticalChalk, name: "Vertical Chalk" },
    { img: VerticalClaystone, name: "Vertical Claystone" },
    { img: VerticalDashed, name: "Vertical Dashed" },
    { img: VerticalDolomiticLimestone, name: "Vertical Dolomitic Limestone" },
    { img: VerticalDolostone, name: "Vertical Dolostone" },
    { img: VerticalFissileMudstone, name: "Vertical Fissile Mudstone" },
    { img: VerticalFissileSiltstone, name: "Vertical Fissile Siltstone" },
    { img: VerticalLimestone, name: "Vertical Limestone" },
    { img: VerticalLines, name: "Vertical Lines" },
    { img: VerticalMarl, name: "Vertical Marl" },
    { img: VerticalShale, name: "Vertical Shale" },
    { img: VerticalTuffitt, name: "Vertical Tuffitt" },
    { img: Vulcanic, name: "Vulcanic" },
]);

export const patternImages = imagesAndNames.map(({ img }) => img);
export const patternNamesEnglish = imagesAndNames.map(({ name }) => name);
