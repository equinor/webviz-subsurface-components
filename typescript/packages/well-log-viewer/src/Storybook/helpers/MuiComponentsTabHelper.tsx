import React from "react";

import type { PayloadAction } from "@reduxjs/toolkit";
import { configureStore, createSlice } from "@reduxjs/toolkit";
import { Provider, useDispatch, useSelector } from "react-redux";

import type { GridDirection } from "@mui/material";
import { Grid } from "@mui/material";

import type { TabItem } from "./MuiComponents";
import { Tabs } from "./MuiComponents";

export interface ITabState {
    value: number;
}

const initialState: ITabState = {
    value: 0,
};

const tabSlices = createSlice({
    name: "tab",
    // `createSlice` will infer the state type from the `initialState` argument
    initialState,
    reducers: {
        setTabValue: (state, action: PayloadAction<number>) => {
            state.value = action.payload;
        },
    },
});

const { setTabValue } = tabSlices.actions;
const getTabValue = (state: TRootState) => state.tab.value;

const store = configureStore({
    reducer: {
        tab: tabSlices.reducer,
    },
});
type TRootState = ReturnType<typeof store.getState>;
type TAppDispatch = typeof store.dispatch;

const renderTestComponent = (
    renderer: (ref: React.MutableRefObject<null>) => React.JSX.Element,
    ref: React.MutableRefObject<null>,
    selectedTabIndex: number
) => {
    // the test component is rendered on tab with index 1
    const testComponentTabIndex = 1;
    if (!ref.current && testComponentTabIndex !== selectedTabIndex) {
        return null;
    } else {
        return renderer(ref);
    }
};

const TestComponentArea: React.FC<React.PropsWithChildren> = ({ children }) => {
    //---------------------------------------------------------------------------------
    // The Page rendering
    //---------------------------------------------------------------------------------

    const direction: GridDirection = "column";

    const ref = React.useRef(null);

    const tab = useSelector(getTabValue);

    const renderer = (ref: React.MutableRefObject<null>): React.JSX.Element => {
        return <div ref={ref}> {children} </div>;
    };

    const testComponent = renderTestComponent(renderer, ref, tab);

    return (
        <div>
            <Grid container direction={direction} justifyContent="flex-start">
                <div>tab title</div>
                {testComponent}
            </Grid>
            <br />
        </div>
    );
};

const AppPage: React.FC<React.PropsWithChildren> = ({ children }) => {
    const renderTabs = (): TabItem[] => {
        const tabItems: TabItem[] = [];

        const renderEmptyTab = (): React.JSX.Element => {
            return <div>empty tab</div>;
        };

        const renderTestComponentTab = (): React.JSX.Element => {
            return <TestComponentArea>{children}</TestComponentArea>;
        };

        const EmptyTab1: TabItem = {
            title: { label: "tab 1" },
            panelContent: renderEmptyTab(),
        };
        tabItems.push(EmptyTab1);

        const logViewTab: TabItem = {
            title: { label: "tab 2" },
            panelContent: renderTestComponentTab(),
        };
        tabItems.push(logViewTab);

        const EmptyTab2: TabItem = {
            title: { label: "tab 3" },
            panelContent: renderEmptyTab(),
        };
        tabItems.push(EmptyTab2);

        return tabItems;
    };

    const dispatch = useDispatch<TAppDispatch>();

    return (
        <Tabs
            name={"blank"}
            tabs={renderTabs()}
            selectedTab={0}
            handleSelectedTab={(tab: number) => dispatch(setTabValue(tab))}
            width="100%"
        />
    );
};

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export const tabDecorator = (Story: any) => (
    <Provider store={store}>
        <AppPage>
            <Story />
        </AppPage>
    </Provider>
);
