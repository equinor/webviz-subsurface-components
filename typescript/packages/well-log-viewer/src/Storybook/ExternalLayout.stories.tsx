import type { Meta, StoryObj } from "@storybook/react";
import React from "react";

import { Grid } from "@mui/material";
//import type { GridDirection } from "@mui/material";

import SyncLogViewer from "../SyncLogViewer";

import { tabDecorator } from "./helpers/MuiComponentsTabHelper";

// eslint-disable-next-line @typescript-eslint/no-var-requires
const args = require("../../../../../example-data/facies3wells.json");

const RTCWellLogViewer: React.FunctionComponent = () => {
    return (
        <Grid container direction={"column"} justifyContent="flex-start">
            <div style={{ width: "100%", height: "100%" }}>
                <div
                    data-testid="well-log-views"
                    style={{ width: "100%", height: "90vh" }}
                >
                    <SyncLogViewer {...{ ...args, id: "c1" }} />
                </div>
            </div>
        </Grid>
    );
};

// storybook page
const meta: Meta<typeof RTCWellLogViewer> = {
    title: "WellLogViewer / Examples / classification layout",
    component: RTCWellLogViewer,
};

export default meta;

export const RTCTabsLayout: StoryObj<typeof RTCWellLogViewer> = {
    decorators: [tabDecorator],
    render: () => (
        <React.StrictMode>
            <RTCWellLogViewer />
        </React.StrictMode>
    ),
};
