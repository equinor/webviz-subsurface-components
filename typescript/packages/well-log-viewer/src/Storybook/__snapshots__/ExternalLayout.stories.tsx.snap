// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WellLogViewer/Demo/ClassificationLayout RTCTabsLayout smoke-test 2`] = `
<div class="MuiBox-root css-0"
     style="width: 100%;"
>
  <div class="MuiTabs-root css-ulxgan-MuiTabs-root">
    <div class="MuiTabs-scrollableX MuiTabs-hideScrollbar css-oqr85h"
         style="width: 99px; height: 99px; position: absolute; top: -9999px; overflow: scroll;"
    >
    </div>
    <div class="MuiTabs-scroller MuiTabs-hideScrollbar MuiTabs-scrollableX css-69z67c-MuiTabs-scroller"
         style="margin-bottom: 0px;"
    >
      <div aria-label="blank-tabs"
           class="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
           role="tablist"
      >
        <button class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary MuiTab-wrapped Mui-selected css-zx9imm-MuiButtonBase-root-MuiTab-root"
                tabindex="0"
                type="button"
                role="tab"
                aria-selected="true"
                id="blank-tab-0"
                aria-controls="blank-tabpanel-0"
                aria-label
                data-mui-internal-clone-element="true"
        >
          <div class="MuiBox-root css-0"
               id="blank-tablabel-0"
          >
            <p class="MuiTypography-root MuiTypography-body2 css-1hb08yz-MuiTypography-root">
              tab 1
            </p>
          </div>
          <span class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root">
          </span>
        </button>
        <button class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary MuiTab-wrapped css-zx9imm-MuiButtonBase-root-MuiTab-root"
                tabindex="-1"
                type="button"
                role="tab"
                aria-selected="false"
                id="blank-tab-1"
                aria-controls="blank-tabpanel-1"
                aria-label
                data-mui-internal-clone-element="true"
        >
          <div class="MuiBox-root css-0"
               id="blank-tablabel-1"
          >
            <p class="MuiTypography-root MuiTypography-body2 css-1hb08yz-MuiTypography-root">
              tab 2
            </p>
          </div>
          <span class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root">
          </span>
        </button>
        <button class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary MuiTab-wrapped css-zx9imm-MuiButtonBase-root-MuiTab-root"
                tabindex="-1"
                type="button"
                role="tab"
                aria-selected="false"
                id="blank-tab-2"
                aria-controls="blank-tabpanel-2"
                aria-label
                data-mui-internal-clone-element="true"
        >
          <div class="MuiBox-root css-0"
               id="blank-tablabel-2"
          >
            <p class="MuiTypography-root MuiTypography-body2 css-1hb08yz-MuiTypography-root">
              tab 3
            </p>
          </div>
          <span class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root">
          </span>
        </button>
      </div>
      <span class="MuiTabs-indicator css-1aquho2-MuiTabs-indicator"
            style="left: 0px; width: 90px;"
      >
      </span>
    </div>
  </div>
  <div role="tabpanel"
       id="tab1-tabpanel"
       data-testid="tab1-tabpanel"
       aria-labelledby="tab1-tab"
  >
    <div>
      empty tab
    </div>
  </div>
  <div role="tabpanel"
       hidden
       id="tab2-tabpanel"
       data-testid="tab2-tabpanel"
       aria-labelledby="tab2-tab"
  >
    <div style="overflow: auto;">
      <div class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-1e5jxqd-MuiGrid-root">
        <div>
          tab title
        </div>
      </div>
      <br>
    </div>
  </div>
  <div role="tabpanel"
       hidden
       id="tab3-tabpanel"
       data-testid="tab3-tabpanel"
       aria-labelledby="tab3-tab"
  >
    <div>
      empty tab
    </div>
  </div>
</div>
`;

exports[`WellLogViewer/Demo/ClassificationLayout RTCTabsLayoutWithScrollbar smoke-test 2`] = `
<div class="MuiBox-root css-0"
     style="width: 100%;"
>
  <div class="MuiTabs-root css-ulxgan-MuiTabs-root">
    <div class="MuiTabs-scrollableX MuiTabs-hideScrollbar css-oqr85h"
         style="width: 99px; height: 99px; position: absolute; top: -9999px; overflow: scroll;"
    >
    </div>
    <div class="MuiTabs-scroller MuiTabs-hideScrollbar MuiTabs-scrollableX css-69z67c-MuiTabs-scroller"
         style="margin-bottom: 0px;"
    >
      <div aria-label="blank-tabs"
           class="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
           role="tablist"
      >
        <button class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary MuiTab-wrapped Mui-selected css-zx9imm-MuiButtonBase-root-MuiTab-root"
                tabindex="0"
                type="button"
                role="tab"
                aria-selected="true"
                id="blank-tab-0"
                aria-controls="blank-tabpanel-0"
                aria-label
                data-mui-internal-clone-element="true"
        >
          <div class="MuiBox-root css-0"
               id="blank-tablabel-0"
          >
            <p class="MuiTypography-root MuiTypography-body2 css-1hb08yz-MuiTypography-root">
              tab 1
            </p>
          </div>
          <span class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root">
          </span>
        </button>
        <button class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary MuiTab-wrapped css-zx9imm-MuiButtonBase-root-MuiTab-root"
                tabindex="-1"
                type="button"
                role="tab"
                aria-selected="false"
                id="blank-tab-1"
                aria-controls="blank-tabpanel-1"
                aria-label
                data-mui-internal-clone-element="true"
        >
          <div class="MuiBox-root css-0"
               id="blank-tablabel-1"
          >
            <p class="MuiTypography-root MuiTypography-body2 css-1hb08yz-MuiTypography-root">
              tab 2
            </p>
          </div>
          <span class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root">
          </span>
        </button>
        <button class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary MuiTab-wrapped css-zx9imm-MuiButtonBase-root-MuiTab-root"
                tabindex="-1"
                type="button"
                role="tab"
                aria-selected="false"
                id="blank-tab-2"
                aria-controls="blank-tabpanel-2"
                aria-label
                data-mui-internal-clone-element="true"
        >
          <div class="MuiBox-root css-0"
               id="blank-tablabel-2"
          >
            <p class="MuiTypography-root MuiTypography-body2 css-1hb08yz-MuiTypography-root">
              tab 3
            </p>
          </div>
          <span class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root">
          </span>
        </button>
      </div>
      <span class="MuiTabs-indicator css-1aquho2-MuiTabs-indicator"
            style="left: 0px; width: 90px;"
      >
      </span>
    </div>
  </div>
  <div role="tabpanel"
       id="tab1-tabpanel"
       data-testid="tab1-tabpanel"
       aria-labelledby="tab1-tab"
  >
    <div>
      empty tab
    </div>
  </div>
  <div role="tabpanel"
       hidden
       id="tab2-tabpanel"
       data-testid="tab2-tabpanel"
       aria-labelledby="tab2-tab"
  >
    <div style="overflow: auto;">
      <div class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-1e5jxqd-MuiGrid-root">
        <div>
          tab title
        </div>
      </div>
      <br>
    </div>
  </div>
  <div role="tabpanel"
       hidden
       id="tab3-tabpanel"
       data-testid="tab3-tabpanel"
       aria-labelledby="tab3-tab"
  >
    <div>
      empty tab
    </div>
  </div>
</div>
`;
